src/ui/MissingTerm/MissingTerms.ts

```diff
@@ -0,0 +1,218 @@
+    this.addMissingTerms();
+  }
+  /**
+   *Returns all original basic query expression terms that were not matched by the result item the component instance is associated with.
+   */
+  public get missingTerms(): string[] {
+    return this.result.absentTerms.filter(term => {
+      const regex = this.createWordBoundryDelimitedRegex(term);
+      return regex.test(this.queryStateModel.get('q'));
+    });
+  }
+
+  /**
+   * Inject a term in the advanced query.
+   */
+  public addTermForcedToAppear(term: string) {
+    if (this.missingTerms.indexOf(term) === -1) {
+      this.logger.warn(
+        'The term to re-inject is not present in the missing terms',
+        `You tried to inject "${term}" but the possible term to inject are: ${this.missingTerms.toString()}`,
+        'The query exited the function'
+      );
+      return;
+    }
+    this.updateTermForcedToAppear();
+    this.termForcedToAppear.push(term);
+    this.queryStateModel.set('missingTerm', [...this.termForcedToAppear]);
+  }
+
+  private updateTermForcedToAppear() {
+    this.termForcedToAppear = [...this.queryStateModel.get('missingTerm')];
+  }
+
+  private addMissingTerms() {
+    if (this.missingTerms.length === 0) {
+      return;
+    }
+    const missingTermelement = this.buildContainer().el;
+    if (missingTermelement) {
+      this.hideMissingTermIfNumberOfResult(missingTermelement);
+      $$(this.element).append(missingTermelement);
+    }
+  }
+
+  private buildContainer(): Dom {
+    const resultCell = $$('div', { className: 'coveo-result-cell' }, this.buildCaption());
+    this.buildMissingTerms().forEach(term => {
+      if (term) {
+        resultCell.append(term.el);
+      }
+    });
+    if (resultCell.el.childElementCount > 1) {
+      return $$('div', { className: 'coveo-result-row' }, resultCell);
+    }
+  }
+
+  private buildCaption(): Dom {
+    return $$('span', { className: 'coveo-field-caption' }, this.options.caption);
+  }
+
+  private buildMissingTerms(): Dom[] {
+    const terms: Dom[] = this.missingTerms.map(term => {
+      if (this.hideMissingTermifFeaturedResults(term) || this.hideMissingTermIfWildcard(term)) {
+        return;
+      }
+      return this.makeTermClickableIfEnabled(term);
+    });
+    return terms;
+  }
+
+  private executeNewQuery(missingTerm: string = this.queryStateModel.get('q')) {
+    this.usageAnalytics.logSearchEvent<IAnalyticsIncludeMissingTerm>(analyticsActionCauseList.missingTermClick, {
+      missingTerm: missingTerm
+    });
+    this.queryController.executeQuery();
+  }
+
+  private makeTermClickableIfEnabled(term: string): Dom {
+    if (this.options.clickable) {
+      const termElement = $$('button', { className: 'coveo-missing-term clickable' }, term);
+      termElement.on('click', () => {
+        this.addTermForcedToAppear(term);
+        this.executeNewQuery(term);
+      });
+      return termElement;
+    } else {
+      return $$('span', { className: 'coveo-missing-term' }, term);
```

Comment:
Similar comment regarding accessibility, even if the term is not clickable, it should be read properly by a screen reader, which a span might not do properly.